JASP-Desktop/widgets/filemenu/filemenu.cpp

```diff
@@ -408,13 +408,33 @@ void FileMenu::setSyncRequest(const QString& path)
 
 bool FileMenu::checkSyncFileExists(const QString &path)
 {
-	if(path.startsWith("http") || (QFileInfo::exists(path) && Utils::getFileSize(path.toStdString()) > 0))
+	if (path.startsWith("http"))
 		return true;
 
-	Log::log() << "Sync file does not exist: " << path.toStdString() << std::endl;
-	clearSyncData();
+	bool exist = false;
 
-	return false;
+	if (QFileInfo::exists(path))
+	{
+		int trials = 0;
+		while (Utils::getFileSize(path.toStdString()) == 0 && trials < 5)
+		{
+			// sometimes it is temporarly empty...
```

Comment:
If you are going to add this weird timer and retry thing, can you make sure we are not wasting time while loading jasp-files?

Reply:
Yes. I split the 2 if statements in 2: first it checks whether the file uberhaupt exists, then it checks whether the file has data, and it loops only if the first condition is met.
I will add comments.